ocaml-lsp-server/src/state.ml

```diff
@@ -64,7 +65,16 @@ let dune t =
 
 let initialize t params workspaces dune =
   assert (t.init = Uninitialized);
-  { t with init = Initialized { params; workspaces; dune } }
+  { t with
+    init =
+      Initialized
+        { params
+        ; workspaces
+        ; dune
+        ; exp_client_caps =
+            lazy (Exp_client_caps.of_opt_json params.capabilities.experimental)
```

Comment:
Doesn't seem write to delay the error checking of this field.

Reply:
I wouldn't say we do any validation. I was under assumption we could compute whether client has experimental capabilities only when user uses ocamllsp functionality that requires those capabilities.